api/metrics/v1/labels_enforcer.go

```diff
@@ -180,33 +132,35 @@ func enforceRequestQueryLabels(e *injectproxy.Enforcer, w http.ResponseWriter, r
 
 // Adapted from
 // https://github.com/prometheus-community/prom-label-proxy/blob/952266db4e0b8ab66b690501e532eaef33300596/injectproxy/routes.go.
-func enforceQueryValues(e *injectproxy.Enforcer, v url.Values) (values string, foundQuery bool, err error) {
-	// If no values were given or no query is present,
-	// e.g. because the query came in the POST body
-	// but the URL query string was passed, then finish early.
-	if v.Get(queryParam) == "" {
-		return v.Encode(), false, nil
-	}
-
-	expr, err := parser.ParseExpr(v.Get(queryParam))
-	if err != nil {
-		return "", true, fmt.Errorf("parse expr error: %w", err)
+func enforceQueryValues(e *injectproxy.Enforcer, paramName string, requestParams url.Values) (values string, foundQuery bool, err error) {
+	if len(requestParams[paramName]) == 0 {
+		enforcedMatchers, err := e.EnforceMatchers([]*labels.Matcher{})
+		if err != nil {
+			return "", false, fmt.Errorf("enforce matchers error: %w", err)
+		}
+		requestParams.Set(paramName, matchersToString(enforcedMatchers...))
+		return requestParams.Encode(), true, nil
 	}
 
-	if err := e.EnforceNode(expr); err != nil {
-		return "", true, fmt.Errorf("enforce node error: %w", err)
+	matchers := requestParams[paramName]
+	for i, rawMatcher := range matchers {
+		expr, err := parser.ParseExpr(rawMatcher)
+		if err != nil {
+			return "", true, fmt.Errorf("parse expr error: %w", err)
+		}
+		if err := e.EnforceNode(expr); err != nil {
+			return "", true, fmt.Errorf("enforce node error: %w", err)
+		}
+		matchers[i] = expr.String()
 	}
-
-	v.Set(queryParam, expr.String())
-
-	return v.Encode(), true, nil
+	requestParams[paramName] = matchers
```

Comment:
Is this needed?

Reply:
Good catch, @thibaultmg. I pushed a272b8ee5651db5c73814b5d1e7016af2f6e5c36 to get rid of it after verifying in a playground (https://go.dev/play/p/NxPYe__Y7Py) it's not required.